sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5321,10 +5321,8 @@ pi_result piKernelGetGroupInfo(pi_kernel Kernel, pi_device Device,
     return ReturnValue(WorkSize);
   }
   case PI_KERNEL_GROUP_INFO_WORK_GROUP_SIZE: {
-    uint32_t X, Y, Z;
-    ZE_CALL(zeKernelSuggestGroupSize,
-            (Kernel->ZeKernel, 10000, 10000, 10000, &X, &Y, &Z));
-    return ReturnValue(size_t{X * Y * Z});
+    return ReturnValue(
+        pi_uint64{Device->ZeDeviceComputeProperties->maxTotalGroupSize});
```

Comment:
The SYCL spec says:

> info::kernel_device_specific::work_group_size | size_t | Returns the maximum number of work-items in a work-group that can be used to execute a kernel on a specific device.

Which seems to be a kernel-specific similar device info:

> info::device::max_work_group_size | size_t | Returns the maximum number of work-items that are permitted in a work-group executing a kernel on a single compute unit. The minimum value is 1.

But the new code here is not kernel-specific, while it should be.
Please elaborate the idea behind the change.

@jandres742 : is there Level Zero API to do what's needed for `info::kernel_device_specific::work_group_size` ?

Reply:
I do agree that I might be somewhat off in using `Device->ZeDeviceComputeProperties->maxTotalGroupSize` as it is just device specific, but using zeKernelSuggestGroupSize  seems to be the wrong implementation since the method returns a suggestion (presumably for optimization purposes) rather than a limit of the size.

Reply:
Please add a comment about L0 missing API for kernel-specific max group size

Reply:
As of right now, L0 is missing API to query kernel and device specific  max work group size. This PR can serve as a temporary workaround while the need for such API is being investigated